immutablecollections/_immutabledict.py

```diff
@@ -66,13 +68,33 @@ def immutabledict(
             f"Cannot create an immutabledict from {type(iterable)}, only {InstantiationTypes}"
         )
 
+    original_length = -1
+    if forbid_duplicate_keys:
+        if not isinstance(iterable, Sized):
+            iterable = list(iterable)
+        original_length = len(iterable)
+
     ret: ImmutableDict[KT, VT] = _RegularDictBackedImmutableDict(iterable)
+
+    if forbid_duplicate_keys and len(ret) != original_length:
+        raise ValueError("iterable has duplicate items and forbid_duplicate_keys=True")
```

Comment:
Rather than just `iterable`, something like
```suggestion
        raise ValueError("Input has duplicate items, but forbid_duplicate_keys=True")
```
where the duplicates are also printed inside the exception.

Reply:
The difficulty is that `list({5:6}) = [5]` but `list([(5,6)]) = [(5,6)]`. So if we want to gather all the keys then we need to branch on the type of input, which is a little annoying. Should we go through with that?

Reply:
Yes, it's worth it.